### ISSUE TITLE
read_html not reading some UTF-8 characters properly from a file

### ISSUE THREAD

**Issue by wkmor1** [r-lib/xml2#293]
In version 1.3.0  some characters are not being read properly from a file

``` r
# xml 1.2.5
tmp <- tempfile()
cat("’", file = tmp)
xml2::read_html(tmp, encoding = "UTF-8")
#> {html_document}
#> <html>
#> [1] <body><p>’</p></body>

# xml 1.3.0
tmp <- tempfile()
cat("’", file = tmp)
xml2::read_html(tmp, encoding = "UTF-8")
#> {html_document}
#> <html>
#> [1] <body><p>â\u0080\u0099</p></body>
```

**Comment by hadley** [comment]
This appears to be breaking pkgdown sites 😞

**Comment by jayhesselberth** [comment]
`git bisect` report:

```
03adb8f5695383270a029128a112ad645bde2df3 is the first bad commit
commit 03adb8f5695383270a029128a112ad645bde2df3
Author: Jim Hester <james.f.hester@gmail.com>
Date:   Fri Mar 27 09:47:17 2020 -0400

    Convert doc_parse_file

 R/RcppExports.R     |  4 ----
 R/xml_parse.R       |  2 +-
 src/RcppExports.cpp | 14 --------------
 src/init.c          |  4 ++--
 src/xml2_doc.cpp    | 31 +++++++++++++++++++------------
 src/xml2_init.cpp   |  2 +-
 6 files changed, 23 insertions(+), 34 deletions(-)
```

**Comment by jimhester** [comment]
Fixed by 654385789

**Comment by jimhester** [comment]
The issue was `strncmp(encoding, "anything", 0)` is always `== 0`, because `strncmp()` only compares up to the minimum number of characters, 0 in this case.

Anyway since C strings are NUL terminated (and therefore always have at least one character) I can just check if `encoding[0] == '\0'`, which is what happens now.
